csharp/src/Ice/LoggerAdmin.cs

```diff
@@ -262,67 +324,45 @@ internal void Destroy()
                     }
                 }
 
-                //
                 // Queue updated, now find which remote loggers want this message
-                //
-                foreach (RemoteLoggerData p in _remoteLoggerMap.Values)
+                foreach (RemoteLoggerQueue p in _remoteLoggerMap.Values)
                 {
-                    Filters filters = p.Filters;
-
-                    if (filters.MessageTypes.Count == 0 || filters.MessageTypes.Contains(logMessage.Type))
+                    if (p.IsAccepted(logMessage))
                     {
-                        if (logMessage.Type != LogMessageType.TraceMessage || filters.TraceCategories.Count == 0 ||
-                           filters.TraceCategories.Contains(logMessage.TraceCategory))
-                        {
-                            if (remoteLoggers == null)
-                            {
-                                remoteLoggers = new List<IRemoteLoggerPrx>();
-                            }
-                            remoteLoggers.Add(p.RemoteLogger);
-                        }
+                        remoteLoggers ??= new List<RemoteLoggerQueue>();
+                        remoteLoggers.Add(p);
                     }
                 }
 
                 return remoteLoggers;
             }
         }
 
-        internal void DeadRemoteLogger(IRemoteLoggerPrx remoteLogger, ILogger logger, System.Exception ex,
-                                       string operation)
-        {
-            //
-            // No need to convert remoteLogger as we only use its identity
-            //
-            if (RemoveRemoteLogger(remoteLogger))
-            {
-                if (_traceLevel > 0)
-                {
-                    logger.Trace(TraceCategory, "detached `" + remoteLogger.ToString() + "' because "
-                                 + operation + " raised:\n" + ex.ToString());
-                }
-            }
-        }
-
-        internal int GetTraceLevel() => _traceLevel;
+        // Change this proxy's communicator
+        private static IRemoteLoggerPrx ChangeCommunicator(IRemoteLoggerPrx prx, Communicator communicator) =>
+            IRemoteLoggerPrx.Parse(prx.ToString()!, communicator);
 
-        private bool RemoveRemoteLogger(IRemoteLoggerPrx remoteLogger)
+        private static Communicator CreateSendLogCommunicator(Communicator communicator, ILogger logger)
         {
-            lock (this)
-            {
-                return _remoteLoggerMap.Remove(remoteLogger.Identity);
-            }
+            var properties = communicator.GetProperties().Where(
+                p => p.Key == "Ice.Default.Locator" ||
+                p.Key == "Ice.Plugin.IceSSL" ||
```

Comment:
There is more to cleanup here. In particular, it would be nice for the communicator to inherits the main communicator's default encoding and default protocol.

Reply:
Fixed with:
```
            var properties = communicator.GetProperties().Where(p =>
                p.Key == "Ice.Default.Locator" ||
                p.Key == "Ice.Default.Encoding" ||
                p.Key == "Ice.Default.Protocol" ||
                p.Key.StartsWith("IceSSL.")).ToDictionary(p => p.Key, p => p.Value);
```